server/golden/proxy.go

```diff
@@ -0,0 +1,467 @@
+	var ok bool
+	conn, ok = h.conns[connId]
+	if ok {
+		return conn, nil
+	} else {
+		conn, err = newConn(h.connStr, connId, h.logger)
+		if err != nil {
+			return proxyConn{}, err
+		}
+	}
+	if err = conn.Ping(); err != nil {
+		return proxyConn{}, err
+	}
+	h.conns[connId] = conn
+	return conn, nil
+}
+
+func (h MySqlProxy) ComInitDB(c *mysql.Conn, schemaName string) error {
+	conn, err := h.getConn(c.ConnectionID)
+	if err != nil {
+		return err
+	}
+	if schemaName != "" {
+		_, err = conn.Exec("USE " + schemaName + " ;")
+	}
+	return err
+}
+
+// ComPrepare parses, partially analyzes, and caches a prepared statement's plan
+// with the given [c.ConnectionID].
+func (h MySqlProxy) ComPrepare(c *mysql.Conn, query string) ([]*querypb.Field, error) {
+	return nil, fmt.Errorf("ComPrepare unsupported")
+}
+
+func (h MySqlProxy) ComStmtExecute(c *mysql.Conn, prepare *mysql.PrepareData, callback func(*sqltypes.Result) error) error {
+	return fmt.Errorf("ComStmtExecute unsupported")
+}
+
+func (h MySqlProxy) ComResetConnection(c *mysql.Conn) {
+	return
+}
+
+// ConnectionClosed reports that a connection has been closed.
+func (h MySqlProxy) ConnectionClosed(c *mysql.Conn) {
+	conn, ok := h.conns[c.ConnectionID]
+	if !ok {
+		return
+	}
+	if err := conn.Close(); err != nil {
+		lgr := logrus.WithField(sql.ConnectionIdLogField, c.ConnectionID)
+		lgr.Errorf("Error closing connection")
+	}
+}
+
+func (h MySqlProxy) ComMultiQuery(
+	c *mysql.Conn,
+	query string,
+	callback func(*sqltypes.Result, bool) error,
+) (string, error) {
+	conn, err := h.getConn(c.ConnectionID)
+	if err != nil {
+		return "", err
+	}
+	conn.Entry = conn.Entry.WithField("query", query)
+
+	remainder, err := h.processQuery(c, conn, query, true, callback)
+	if err != nil {
+		conn.Errorf("Failed to process MySQL results: %s", err)
+	}
+	return remainder, err
+}
+
+// ComQuery executes a SQL query on the SQLe engine.
+func (h MySqlProxy) ComQuery(
+	c *mysql.Conn,
+	query string,
+	callback func(*sqltypes.Result, bool) error,
+) error {
+	conn, err := h.getConn(c.ConnectionID)
+	if err != nil {
+		return err
+	}
+	conn.Entry = conn.Entry.WithField("query", query)
+
+	_, err = h.processQuery(c, conn, query, false, callback)
+	if err != nil {
+		conn.Errorf("Failed to process MySQL results: %s", err)
+	}
+	return err
+}
+
+func (h MySqlProxy) processQuery(
+	c *mysql.Conn,
+	proxy proxyConn,
+	query string,
+	isMultiStatement bool,
+	callback func(*sqltypes.Result, bool) error,
+) (string, error) {
+	ctx := sql.NewContext(h.ctx)
+	var remainder string
+	if isMultiStatement {
+		_, query, remainder, _ = parse.ParseOne(ctx, query)
+	}
+
+	ctx = ctx.WithQuery(query)
+	more := remainder != ""
+
+	proxy.Debugf("Sending query to MySQL")
+	rows, err := proxy.Query(query)
+	if err != nil {
+		return "", err
+	}
+	defer func() {
+		if cerr := rows.Close(); cerr != nil {
+			err = cerr
+		}
+	}()
+
+	var proccesedAtLeastOneBatch bool
```

Comment:
You spelled it right in the comment at the bottom of the function 🤣 
```suggestion
	var processedAtLeastOneBatch bool
```